### DISCUSSION TITLE
Dokku Deployment Configuration

### DISCUSSION THREAD

**Discussion by barelyhuman** [vercel/vercel#4530] [Help] [answered]
So I've got a simple nextjs project with api routes and pages as standardized. 
the thing is I'm using axios with a baseURL for localhost:{process.env.PORT}/api/v1 for post and get request from the client rendered pages.

this works fine on local system but dokku creates a container with a dynamic port for each app 
the server now runs on localhost:5000 but the app can't find the PORT variable anymore , making it unusable,
how do you suggest to get it to work in such cases?

**Edit** : The issue lies with serverSideProps as it's unable to use absolute paths, compared to client rendered where it can just append request to the actual url.

**Comment by barelyhuman** [comment]
I ended up going with the approach I didn't want too. 

```js
const getBaseURL = () => {
  let url;
  if (typeof window === 'undefined') {
    process.env.PORT = process.env.PORT || 3000;
    url = `http://localhost:${process.env.PORT}/api/v1`;
  } else {
    url = `${window.location.origin}/api/v1/`;
  }
  return url;
};
```

make sure that axios calls the function to dynamically get the accessible route

**Comment by coetry** [comment, marked as answer]
This support channel is for Vercel deployments not for Dokku :smile: . You can also raise your question to the [Next.js community](https://github.com/vercel/next.js/discussions) since people adopt several different deployment strategies alongside Vercel.

**Reply by barelyhuman** [reply]
My bad, i'll post it up there
